### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by anngy12**

### CODE REVIEW

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <div class="d-flex justify-content-between small">
+                <span>@MinPrice€</span>
+                <span>@MaxPrice€</span>
+            </div>
+            <input type="range" class="form-range" id="priceSlider">
+        </div>
+    </div>
+</form>
+
+@code {
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:44]: 🔧Could you press "Alt-Str-L" to format the file?

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+// Keywords for the Filter
+private List<string> Keywords { get; set; } = new List<string> { "Keyword", "Key", "Word" };
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:47]: ⛏️ You could use a collection expression
```suggestion
private List<string> Keywords { get; set; } = ["Keyword", "Key", "Word"];
```

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    public string Description { get; set; }
+}
+
+// List of CheckboxItem Objects
+private List<CheckboxItem> CheckboxItems { get; set; } = new List<CheckboxItem>
+{
+new CheckboxItem { Label = "Label", Description = "Description" },
+new CheckboxItem { Label = "Label", Description = "Description" },
+new CheckboxItem { Label = "Label", Description = "Description" }
+};
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:67]: ⛏️ Here the same
```suggestion
private List<CheckboxItem> CheckboxItems { get; set; } =
[
new CheckboxItem { Label = "Label", Description = "Description" },
new CheckboxItem { Label = "Label", Description = "Description" },
new CheckboxItem { Label = "Label", Description = "Description" }
];
```

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+{
+Keywords.Remove(keyword);
+}
+
+// Checkbox Item Class
+private class CheckboxItem
+{
+    public string Label { get; set; }
+    public string Description { get; set; }
+}
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:59]: 🔧The Attributes of the class could be null at the end of the Constructor. Maybe we should require them.
```suggestion
private class CheckboxItem
{
    public require string Label { get; set; }
    public require string Description { get; set; }
}
```

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    <div class="p-3 bg-light rounded" style="width: 250px;">
+
+
+        <!-- Keyword Tags -->
+        <h6>Keywords</h6>
+        <div class="d-flex flex-wrap gap-2 mb-3">
+            @foreach (var keyword in Keywords)
+            {
+                <span class="badge bg-secondary">
+                    @keyword <button class="btn-close btn-close-white btn-sm ms-2-5" @onclick="() => RemoveKeyword(keyword)"></button>
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:15]: 🤔 For shorter line width we could think about something like
```suggestion
                    @keyword 
                    <button class="btn-close btn-close-white btn-sm ms-2-5" 
                            @onclick="() => RemoveKeyword(keyword)">
                    </button>
```

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                <input class="form-check-input" type="checkbox" value="" id="@item.Label">
+                <label class="form-check-label" for="@item.Label">
+                    <strong>@item.Label</strong>
+                    <div class ="small text-muted"> @item.Description</div>
+                </label>
+            </div>
+        }
+        <!-- Slider -->
+        <div class="mt-3">
+            <label for="priceSlider" class="form-label"><strong>Label</strong></label>
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:34]: 🔧 Lets not nest multiple HTML elements in one line
```suggestion
            <label for="priceSlider" 
                   class="form-label">
                <strong>Label</strong>
            </label>
```

**Review by Sc3l3t0n — CHANGES_REQUESTED** [review]

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 7]
```diff
@@ -25,13 +27,15 @@
                 <input class="form-check-input" type="checkbox" value="" id="@item.Label">
                 <label class="form-check-label" for="@item.Label">
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:28]: 🔧 Can you move the attributes in their own line too?

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 8]
```diff
@@ -25,13 +27,15 @@
...
                     <strong>@item.Label</strong>
-                    <div class ="small text-muted"> @item.Description</div>
+                    <div class="small text-muted"> @item.Description</div>
                 </label>
             </div>
         }
         <!-- Slider -->
         <div class="mt-3">
-            <label for="priceSlider" class="form-label"><strong>Label</strong></label>
+            <label for="priceSlider" class="form-label">
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:36]: 🔧 Can you move the attributes in their own line too?

**Review by Sc3l3t0n — CHANGES_REQUESTED** [review]
After this I'm completely happy

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 9]
```diff
@@ -0,0 +1,77 @@
+﻿@* FilterComponent *@
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:1]: Sorry for the inconvenience, but you need to change the name of the component and file to `Filter` like I described in #21

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+        <!-- Keyword Tags -->
+        <h6>Keywords</h6>
+        <div class="d-flex flex-wrap gap-2 mb-3">
+            @foreach (var keyword in Keywords)
+            {
+                <span class="badge bg-secondary">
+                    @keyword
+                    <button class="btn-close btn-close-white btn-sm ms-2"
+                            @onclick="() => RemoveKeyword(keyword)">
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:16]: 🔧 Sorry, I made a mistake with the formatting. We can format it like this, but the way it's mentioned in #19 is easier to read IMO
```suggestion
                    <button 
                        class="btn-close btn-close-white btn-sm ms-2"
                        @onclick="() => RemoveKeyword(keyword)"
                    >
```

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+                </span>
+            }
+        </div>
+
+
+        <!-- Checkbox List -->
+        @foreach (var item in CheckboxItems)
+        {
+            <div class="form check mb-2">
+                <input class="form-check-input" type="checkbox" value="" id="@item.Label">
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:27]: 🔧 Like the above

File: src/TuDa.CIMS.Web/Components/FilterComponent.razor [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        <!-- Slider -->
+        <div class="mt-3">
+            <label for="priceSlider" class="form-label">
+                <strong>Label</strong>
+            </label>
+            <div class="d-flex justify-content-between small">
+                <span>@MinPrice€</span>
+                <span>@MaxPrice€</span>
+            </div>
+            <input type="range" class="form-range" id="priceSlider">
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/FilterComponent.razor:43]: 🔧 Like the above

**Review by Sc3l3t0n — CHANGES_REQUESTED** [review]
Sorry for the thing with #19 and #21
After this, I'm happy with everything 👍
Looks nice btw👀

**Comment by Sc3l3t0n** [comment]
Approved

File: src/TuDa.CIMS.Web/Components/Filter.razor
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/Filter.razor:1]: :wrench: I know I'm going round in circles, but the formatting I wanted is not supported for razor files in Rider. So please run format again, and then I'm finally happy :sob:. Sorry for this mess :hankey:

Comment (anngy12) [reply src/TuDa.CIMS.Web/Components/Filter.razor:1]: what do you mean with run format again? STRG + ALT + L maybe?

**Review by Sc3l3t0n — APPROVED** [review]

**Review by devidsdt — APPROVED** [review]

**Pull request merged**